rim/tests.py

```diff
@@ -1,3 +1,43 @@
-from django.test import TestCase
+from django.test import TestCase, Client, RequestFactory, SimpleTestCase
+from django.contrib.auth.models import AnonymousUser, User
+from django.urls import reverse
+from rim import urls
 
 # Create your tests here.
+class TestHttpResponse(TestCase):
+    fixtures = ['dummydb.json']
+    def setUp(self):
+        
+        # Url dictionary to hold names for existing rim web pages, add as neccessary 
+        self.urlDictionary = {
+            "Home": reverse('home'),
+            "Add": reverse('add'),
+            "Edit": reverse('edit', args = [1]),
+            "Client": reverse('client', args = [1]),
+            "Client List": reverse('client_list')
+        }
+        
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(
+            username='test', email='', password='test')
+        self.c = Client()
```

Comment:
FYI: There's already a `self.client` object. All should need to do is the `self.client.login()` - Django does everything else for you.